File: ClientApp/src/app/models/student.ts
export class Student {
    id: number;
    firstName: string;
    lastName: string;
    username: string;
    email: string;
    collegeName: string;
    fkCollegeId: number;
    course: string;
    yearOfStudy: number;
    resumeBase64: string;
    createdAt?: Date;
    updatedAt?: Date;

    constructor() {
        this.id = 0;
        this.firstName = '';
        this.lastName = '';
        this.username = '';
        this.email = '';
        this.collegeName = '';
        this.fkCollegeId = 0;
        this.course = '';
        this.yearOfStudy = 0;
        this.resumeBase64 = '';
        this.createdAt = undefined;
        this.updatedAt = undefined
    }
}

File: Intern.Models/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Intern.Models
{
    public class Student
    {
        public int Id { get; set; }
        public int FkUserId { get; set; }
        [JsonIgnore]
        public User User{ get; set; }
        public int FkCollegeId { get; set; }
        public string Course { get; set; }
        public int YearOfStudy { get; set; }
        public string? ResumeBase64 { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }
    }
}
